src/http/header.c

```diff
@@ -94,17 +94,22 @@ void header_insert_move(Header* restrict h, SSPair* restrict p) {
 }
 
 void* header_get(const Header* restrict h, const char* restrict key) {
-	if(unlikely(!h || !key || *key))
-		return NULL;
+	if(unlikely(h == NULL))
```

Comment:
Well, I think the original condition is okay.

Reply:
The third condition should be !(*key) then.
Combining booleans together will easily cause bug like this and hard to find.

Reply:
If key is NULL, it will trigger the short cut and enter this segment.

Reply:
If `key` is not `NULL` and it's not empty string (like `"\0"`), `*key` will not be 0.
Which means the condition is true, the segment is triggered, and it returns NULL.

Reply:
Obviously, yes.

Reply:
Oh yes, it is a bug.